mail_activity_team/models/mail_activity.py

```diff
@@ -27,6 +27,19 @@ def _get_default_team_id(self, user_id=None):
         comodel_name="mail.activity.team", default=lambda s: s._get_default_team_id()
     )
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            # ensure that the built-in user_id field value is in sync with our field
+            # team_user_id before triggering create
+            if (
+                "user_id" in vals
+                and "team_user_id" in vals
+                and vals["user_id"] != vals["team_user_id"]
+            ):
+                vals["user_id"] = vals["team_user_id"]
```

Comment:
```suggestion
        vals_list = [
            # ensure that the built-in user_id field value is in sync with our field
            # team_user_id before triggering create
            dict(vals, user_id=vals["team_user_id"]) if (
                "user_id" in vals
                and "team_user_id" in vals
                and vals["user_id"] != vals["team_user_id"]
            ) else vals
            for vals in vals_list
        ]
```

(never change your input, make a copy if you're going to change it)

Reply:
Thank you for the tip